frontend/src/components/annotation/annotation_core.vue

```diff
@@ -3886,45 +3921,42 @@ mplate_has_keypoints_type: function (instance_template) {
       }
 
       // Set Metadata to manage saving frames
-      if (this.instance_buffer_metadata[frame_number]) {
-        this.instance_buffer_metadata[frame_number].pending_save = true;
-      } else {
-        this.instance_buffer_metadata[frame_number] = { pending_save: true };
-      }
+      this.set_frame_pending_save(true, frame_number)
     },
 
     // TODO rename? / refactor? in contect of more awareness of ref/by value for buffer
+    add_instance_to_file: async function(instance, frame_number = undefined){
+      if(this.video_mode){
+        if(frame_number == undefined){
+          console.error('Please provide a frame number to call add_instance_to_file()')
+          return
+        }
+        this.add_instance_to_frame_buffer(instance, frame_number)
+      }
+      else{
+        this.push_instance_to_image_file(instance, frame_number)
```

Comment:
Can we get rid of `frame_number` entirely for this function? or is there a directional reason to keep it?

Reply:
Thanks Anthony! Yes good catch :) I already removed it here: https://github.com/diffgram/diffgram/pull/491/commits/db6c15a034e003a9f63889cbf701105f4c4bc869

Reply:
Thanks Pablo! :)